code/parse/sexp_container.cpp

```diff
@@ -329,6 +337,233 @@ sexp_container *get_sexp_container_special(const SCP_string& text, size_t start_
 	return nullptr;
 }
 
+/**
+* Helper function for sexp_container_replace_refs_with_values(). Given a SEXP Container it works out what modifer was used and what the replacement string should be.
+**/
+bool get_replace_text_for_modifier(const SCP_string &text,
+	sexp_container &container,
+	const size_t lookHere,
+	SCP_string &replacement_text,
+	size_t &num_chars_to_replace)
+{
+	// for map containers, check if this matches a map key
+	if (container.is_map()) {
+		if (container.map_data.empty()) {
+			Warning(LOCATION, "Attempt to replace text using empty map container %s", container.container_name.c_str());
```

Comment:
The FREDder can Ctrl+F in Notepad++ for the string "&<MapContainerName>&" where `<MapContainerName>` is the name of the map container that appears in the Warning. They then need to figure out why this map container is empty at the time this text gets replaced.

Since the warning is about text replacement, it's unrelated to SEXPs but rather is about player-facing text in some briefing/message/debriefing/subtitle/etc.

I'm not sure how to word this message to make it clearer to FREDders what they did wrong. I don't think the code has enough context at this point to tell the FREDder which message/briefing/etc. it was. Any ideas?

All that the player can do is report it, so that the FREDder knows their mission has a bug.

BTW, `Warning()` messages get written to the debug log, right?

Reply:
I think warning does get to the debug log, but @Goober5000 can you confirm?

Reply:
Now that I fully understand the context, I think you're right that there isn't much we can do to clarify unless we want to spoon-feed them with a "Make sure the container has text before trying to use it for replacements!"

Reply:
Belatedly -- yes, warnings are written to the debug log.